server/db/db.js

```diff
@@ -4,123 +4,190 @@
-    _verbose('db.getVisits', arguments);
+  get: function(fxaId, count, cb) {
+    _verbose('db.visits.get', fxaId, count);
     var query = 'SELECT id, url, urlHash, title, visitedAt ' +
-                'FROM visits WHERE fxaId = ? ORDER BY visitedAt DESC LIMIT ?';
-    pool.query(query, [fxaId, count], function(err, results) {
-      cb(err, results);
+                'FROM visits WHERE fxaId = $1 ORDER BY visitedAt DESC LIMIT $2';
+    pg.connect(dbParams, function(err, client, done) {
+      client.query(query, [fxaId, count], function(err, results) {
+        if (err) {
+          log.warn('error getting visits: ' + err);
+        } else {
+          _verbose('db.visits.get succeeded');
+        }
+        done();
+        cb(err, results && results.rows);
+      });
     });
-  },
-  getVisit: function(fxaId, visitId, cb) {
-    _verbose('db.getVisit', arguments);
-    // important: MySQL enforces that the user with id `fxaId` is the user with visit `visitId`
+  }
+};
+
+var visit = {
+  get: function(fxaId, visitId, cb) {
+    _verbose('db.visit.get', fxaId, visitId);
+    // important: postgres enforces that the user with id `fxaId` is the user with visit `visitId`
     var query = 'SELECT id, url, urlHash, title, visitedAt ' +
-                'FROM visits WHERE id = ? AND fxaId = ?';
-    pool.query(query, [visitId, fxaId], function(err, r) {
-      if (err) {
-        log.warn('error getting visit: ' + err);
-      }
-      cb(err, r && r[0]);
+                'FROM visits WHERE id = $1 AND fxaId = $2';
+    pg.connect(dbParams, function(err, client, done) {
+      client.query(query, [visitId, fxaId], function(err, r) {
+        if (err) {
+          log.warn('error getting visit: ' + err);
+        } else {
+          _verbose('db.visits.get succeeded');
+        }
+        done();
+        cb(err, r && r.rows[0]);
+      });
     });
   },
   // TODO if the url and visitedAt are the same, should we just discard the record?
   // ...maybe just deal with it later
-  createVisit: function(fxaId, visitId, visitedAt, url, title, cb) {
-    _verbose('db.createVisit', arguments);
-    var urlHash = createHash('sha1').update(url).digest('hex').toString();
+  create: function(fxaId, visitId, visitedAt, url, title, cb) {
+    _verbose('db.visit.create', fxaId, visitId, visitedAt, url, title);
+    var urlHash = crypto.createHash('sha1').update(url).digest('hex').toString();
     var query = 'INSERT INTO visits (id, fxaId, rawUrl, url, urlHash, title, visitedAt) ' +
-                'VALUES (?, ?, ?, ?, ?, ?, ?)';
-    pool.query(query, [visitId, fxaId, url, url, urlHash, title, visitedAt], function(err, results) {
-      if (err) {
-        log.warn('error creating visit: ' + err);
-      }
-      cb(err);
+                'VALUES ($1, $2, $3, $4, $5, $6, $7)';
+    pg.connect(dbParams, function(err, client, done) {
+      client.query(query, [visitId, fxaId, url, url, urlHash, title, visitedAt], function(err, r) {
```

Comment:
So, we're inserting the URL here as both `rawUrl` and `url`. Do we want to be normalizing one of those or something?
I saw this via npm tweets the other day, not sure if it's useful: https://github.com/sindresorhus/normalize-url

Or, if we want to wrap our own util function around URL normalization &mdash; so we can strip out certain URL query params (e.g. Google analytics tracking codes, etc).

Or, tell me to shut up and we can kick this can far, far down the road.